### DISCUSSION TITLE
DataDog scaler creates pods but with Terminating status

### DISCUSSION THREAD

**Discussion by raviumishra** [kedacore/keda#5996] [Q&A / Need Help]
`apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: datadog-scaledobject
  namespace: dev-keda-test-ns
spec:
  scaleTargetRef:
    name: keda-test
  minReplicaCount: 1
  maxReplicaCount: 3
  triggers:
  - type: datadog
    metadata:
      query: "sum:nginx.net.request_per_s{*}" 
      queryValue: "36"
    authenticationRef:
      name: datadog-trigger-auth
`
this scales KEDA but pods are spun with Terminating status
![image](https://github.com/user-attachments/assets/c4b3e99d-9e68-4648-846b-862ff97f7271)
![image](https://github.com/user-attachments/assets/f43f950f-0c8b-40f8-b55e-808eec0b2b5c)

**Comment by arapulido** [comment]
Hello @raviumishra. The Datadog scaler or KEDA do not scale the object, HPA does. If a pod is Terminating, is something transitional. A pod will either be created and running, or teminated (if the value is below your target, as it seems it is).

**Reply by raviumishra** [reply]
Hello @arapulido. I have changed my target value for scaling up as 16 nginx.net.request_per_s. Attached is the historical metric data and also the snapshot shows that at any point of time the  metric is above 16. Still the pods are being spun up with Terminating status.
![image](https://github.com/user-attachments/assets/133009ec-e1a4-492a-811f-b93c1419ae73)
[nginx_req_s.csv](https://github.com/user-attachments/files/16427074/nginx_req_s.csv)
![image](https://github.com/user-attachments/assets/56d0a54c-1388-444b-8f13-6e806c570412)
![image](https://github.com/user-attachments/assets/2e0e5b39-a464-48b0-8ea9-0cbf01e2e823)

![image](https://github.com/user-attachments/assets/d996dfeb-dc30-4cfa-9631-cd476d20ba7c)

**Comment by arapulido** [comment]
Hello! KEDA nor Datadog Scaler perform the actual creation of the pods, the ReplicaSet controller does.

to debug this, I would recommend you to `kubectl describe` both the ReplicaSet, and the terminating pods

**Reply by raviumishra** [reply]
@arapulido  
![image](https://github.com/user-attachments/assets/c59b21de-9e54-41c4-bf47-8566e5a4cc54)
The pods are created with Terminating status -- meaning they just appear and disappear very quickly. Not getting a window to describe them.

https://github.com/user-attachments/assets/ed11e064-eb42-4f62-b4c5-918fc6075fe7

**Comment by arapulido** [comment]
Can you also describe the HPA object?

**Reply by raviumishra** [reply]
@arapulido 
![image](https://github.com/user-attachments/assets/b2a2ff7b-8231-42b2-9c26-3e320b711a8a)

**Comment by arapulido** [comment]
Can you ensure the screenshots show all events? Also, it would be useful to describe the Deployment object as well.

**Reply by raviumishra** [reply]
@arapulido Yes, all events are captured in these screenshots.
![image](https://github.com/user-attachments/assets/5ffb99c4-e822-4a48-8d71-202f91bec5ca)

![image](https://github.com/user-attachments/assets/03933cc0-519d-458b-8af2-03d2ec032e11)

**Comment by arapulido** [comment]
Without having access to your environment it is going to be difficult for me to know what's going on. This requires live debugging. Either KEDA is scaling up and down, or something else. Without access to your logs, I am afraid I am not able to help

**Reply by raviumishra** [reply]
@arapulido Is there a way we can connect for this OR can you pls give me some debugging steps ?

**Reply by zroubalik** [reply]
I am happy to help, let's connect.
